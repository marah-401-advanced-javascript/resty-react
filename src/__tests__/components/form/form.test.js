import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../../../components/form/form';

describe('<Form>', () => {

  it('is alive at application start', () => {
    const form = shallow(<Form />);
    expect(form.find('main form div input').exists()).toBeTruthy();
    expect(form.find('.method').exists()).toBeTruthy();
    expect(form.find('.results').exists()).toBeTruthy();
  });

  it('properly store the users input into state', () => {
    const form = mount(<Form />);
    const urlInput = form.find('.input');
    const methodInput = form.find('.methodGet');
    urlInput.simulate('change',{ target: { value: 'url' } });
    methodInput.simulate('click');
    expect(form.state('url')).toEqual('url');
    expect(form.state('method')).toEqual('get');
  });

  it('properly clear the form/state after the form is submitted', () => {
    const form = mount(<Form />);
    const button = form.find('button');
    button.simulate('submit');
    expect(form.state('method')).toBeFalsy();
    expect(form.state('url')).toBeFalsy();
  });

  it('properly display the output on form submit', () => {
    const form = mount(<Form />);
    const button = form.find('button');
    const urlInput = form.find('.input');
    const methodInput = form.find('.methodGet');
    urlInput.simulate('change',{ target: { value: 'url' } });
    methodInput.simulate('click');
    button.simulate('submit');
    expect(form.find('.methodOutput').text()).toContain('get');
    expect(form.find('.urlOutput').text()).toContain('url');
  });

  it('Renders correctly', () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});