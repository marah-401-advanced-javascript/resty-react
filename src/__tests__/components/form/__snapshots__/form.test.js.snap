// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form> Renders correctly 1`] = `
<main
  className="main"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="div"
    >
      <input
        className="input"
        onChange={[Function]}
        placeholder="Enter URL"
        type="text"
      />
      <button
        type="submit"
      >
        GO!
      </button>
    </div>
    <div
      className="div"
    >
      <span
        className="method"
      >
        <input
          className="methodGet"
          name="method"
          onClick={[Function]}
          type="radio"
          value="get"
        />
        GET
      </span>
      <span
        className="method"
      >
        <input
          name="method"
          onClick={[Function]}
          type="radio"
          value="post"
        />
        POST
      </span>
      <span
        className="method"
      >
        <input
          name="method"
          onClick={[Function]}
          type="radio"
          value="put"
        />
        PUT
      </span>
      <span
        className="method"
      >
        <input
          name="method"
          onClick={[Function]}
          type="radio"
          value="delete"
        />
        DELETE
      </span>
    </div>
  </form>
  <div
    className="results"
  >
    <h3
      className="methodOutput"
    />
    <h3
      className="urlOutput"
    >
       
    </h3>
  </div>
</main>
`;
